### PR TITLE
modify defining group functions to use perm group instead of pc group

### PR DESCRIPTION
**Pull request by drallenherman**
Modify DefiningGroup functions to use PermGroup instead of SpecialPcGroup to improve performance

### CODE REVIEW

**Comment by olexandr-konovalov** [comment]
@drallenherman what this was? need help?

File: lib/div-alg.gi [hunk 1]
```diff
@@ -394,7 +394,9 @@ if (l=4) then
   g:=f/[a^A[3],b^A[4][1]*a^(-A[4][3]),b^(-1)*a*b*a^(-A[4][2])];
 fi;
 
-I:=IsomorphismSpecialPcGroup(g);
+##### Change to Permutation Group for performance ####
```
Comment (olexandr-konovalov) [inline lib/div-alg.gi:397]: This commit seems the only change that should be here - was the problem that you submitted it into master, not into divalg branch?

Comment (drallenherman) [reply lib/div-alg.gi:397]: Same change in two places.  When I posted it on github I had the same problem as two days ago my computer said it was based on github with only one commit ahead of divalg but when I pushed it to github its page showed it based on master with thirty something commits.  So I just closed and deleted it.  I am trying to push a pr/73 using git bash now but its not letting me push as there is no existing pr/73 there yet.

Comment (olexandr-konovalov) [reply lib/div-alg.gi:397]: That was easy to fix - I restored closed branch, reopened this PR, and changed its base branch - so now it looks like being submitted to divalg, where it should go.

File: lib/div-alg.gi [hunk 2]
```diff
@@ -504,7 +506,9 @@ b:=f.2;
 g:=f/[a^A[3],b^A[4][1]*a^(-A[4][3]),b^(-1)*a*b*a^(-A[4][2])];
 fi;
 
-I:=IsomorphismSpecialPcGroup(g);
+### Change to Permutation Group for performance ###
+# I:=IsomorphismSpecialPcGroup(g);
```
Comment (olexandr-konovalov) [inline lib/div-alg.gi:510]: We need to stop commenting out a block of code and writing a new code below. This makes diffs unclear, whereas editing IN the code makes changes more obvious. if you need to preserve commented out code, I've rely on version control to keep its previous version.

**Comment by olexandr-konovalov** [comment]
Tests are in. This may not be necessarily a good thing:
```
########> Diff in /home/travis/build/gap-packages/wedderga/gaproot/pkg/wedderg\
a/tst/wedderga07.tst:217
# Input is:
LocalIndicesOfCyclotomicAlgebra(W[27]);
# Expected output:
[ [ infinity, 2 ] ]
# But found:
Error, reached the pre-set memory limit
(change it with the -o command line option)
########
```
perhaps some heuristics depending on the properties of the group are needed?

**Comment by drallenherman** [comment]
> Tests are in. This may not be necessarily a good thing:
> 
> ```
> ########> Diff in /home/travis/build/gap-packages/wedderga/gaproot/pkg/wedderg\
> a/tst/wedderga07.tst:217
> # Input is:
> LocalIndicesOfCyclotomicAlgebra(W[27]);
> # Expected output:
> [ [ infinity, 2 ] ]
> # But found:
> Error, reached the pre-set memory limit
> (change it with the -o command line option)
> ########
> ```
> 
> perhaps some heuristics depending on the properties of the group are needed?

I do not understand why this happens.  LocalIndexAtTwoByCharacter isn't working on this algebra, because at some point in the middle after the field is increased to the PSplitSubextension, when it tries to do SimpleComponentOfGroupRingByCharacter, GAP closes the window like what happens in a memory crash at that step.  DefiningGroup returns a result almost immediately.   I don't see how switching to a Permutation Group can cause the problem here and not with divalg, but I don't know how this change might interact with the change to SimpleAlgebraInfoByData that Angel made.

**Comment by angeldelriomateos** [comment]
> Tests are in. This may not be necessarily a good thing:
> 
> ```
> ########> Diff in /home/travis/build/gap-packages/wedderga/gaproot/pkg/wedderg\
> a/tst/wedderga07.tst:217
> # Input is:
> LocalIndicesOfCyclotomicAlgebra(W[27]);
> # Expected output:
> [ [ infinity, 2 ] ]
> # But found:
> Error, reached the pre-set memory limit
> (change it with the -o command line option)
> ########
> ```
> 
> perhaps some heuristics depending on the properties of the group are needed?

Can you give more information about what is the input in this calculation so that I can reproduce it in my computer and invetigate what is the problem?

**Comment by drallenherman** [comment]
> > Tests are in. This may not be necessarily a good thing:
> > ```
> > ########> Diff in /home/travis/build/gap-packages/wedderga/gaproot/pkg/wedderg\
> > a/tst/wedderga07.tst:217
> > # Input is:
> > LocalIndicesOfCyclotomicAlgebra(W[27]);
> > # Expected output:
> > [ [ infinity, 2 ] ]
> > # But found:
> > Error, reached the pre-set memory limit
> > (change it with the -o command line option)
> > ########
> > ```
> > 
> > 
> > perhaps some heuristics depending on the properties of the group are needed?
> 
> Can you give more information about what is the input in this calculation so that I can reproduce it in my computer and invetigate what is the problem?

LocalIndices on last component of QG for SG(480,600).  divalg does it no problem.  Using pr/72 with PermGroup replacing SpecialPcGroup in DefiningGroup and DefiningGroupAndCharacter functions (no other changes, and this is usually faster), there is a memory overload on one of the SimpleComponentOfGroupRingByCharacter steps in LocalIndexAtTwoByCharacter.  It makes no sense.

**Comment by angeldelriomateos** [comment]
I don't have problems with this calculation:

gap> G:=SmallGroup(480,600);
<pc group of size 480 with 7 generators>
gap> QG:=GroupRing(Rationals,G);
<algebra-with-one over Rationals, with 7 generators>
gap> wd := WedderburnDecompositionInfo(QG);
[ [ 1, Rationals ], [ 1, Rationals ], [ 1, Rationals ], [ 1, Rationals ], [ 1, Rationals ], 
  [ 1, Rationals ], [ 1, Rationals ], [ 1, Rationals ], [ 2, Rationals ], [ 2, Rationals ], 
  [ 2, Rationals ], [ 2, NF(5,[ 1, 4 ]) ], [ 2, Rationals ], [ 2, Rationals ], [ 2, Rationals ], 
  [ 2, NF(5,[ 1, 4 ]) ], [ 2, NF(5,[ 1, 4 ]) ], [ 2, NF(5,[ 1, 4 ]) ], [ 2, Rationals, 4, [ 2, 3, 2 ] ], 
  [ 4, Rationals ], [ 4, NF(5,[ 1, 4 ]) ], [ 4, NF(5,[ 1, 4 ]) ], 
  [ 1, NF(24,[ 1, 5, 7, 11 ]), 24, [ [ 2, 5, 18 ], [ 2, 7, 12 ] ], [ [ 6 ] ] ], [ 4, NF(5,[ 1, 4 ]) ], 
  [ 1, NF(40,[ 1, 11, 29, 39 ]), 40, [ [ 2, 11, 20 ], [ 2, 29, 30 ] ], [ [ 10 ] ] ], 
  [ 4, NF(15,[ 1, 14 ]) ], [ 1, NF(5,[ 1, 4 ]), 60, [ [ 2, 11, 0 ], [ 2, 19, 30 ], [ 2, 31, 30 ] ], 
      [ [ 0, 45 ], [ 15 ] ] ] ]
gap> A:=wd[27];
[ 1, NF(5,[ 1, 4 ]), 60, [ [ 2, 11, 0 ], [ 2, 19, 30 ], [ 2, 31, 30 ] ], [ [ 0, 45 ], [ 15 ] ] ]
gap> LocalIndicesOfCyclotomicAlgebra(A);
[ [ infinity, 2 ] ]

**Comment by drallenherman** [comment]
> I don't have problems with this calculation:
> 
> gap> G:=SmallGroup(480,600);
> <pc group of size 480 with 7 generators>
> gap> QG:=GroupRing(Rationals,G);
> <algebra-with-one over Rationals, with 7 generators>
> gap> wd := WedderburnDecompositionInfo(QG);
> [ [ 1, Rationals ], [ 1, Rationals ], [ 1, Rationals ], [ 1, Rationals ], [ 1, Rationals ],
> [ 1, Rationals ], [ 1, Rationals ], [ 1, Rationals ], [ 2, Rationals ], [ 2, Rationals ],
> [ 2, Rationals ], [ 2, NF(5,[ 1, 4 ]) ], [ 2, Rationals ], [ 2, Rationals ], [ 2, Rationals ],
> [ 2, NF(5,[ 1, 4 ]) ], [ 2, NF(5,[ 1, 4 ]) ], [ 2, NF(5,[ 1, 4 ]) ], [ 2, Rationals, 4, [ 2, 3, 2 ] ],
> [ 4, Rationals ], [ 4, NF(5,[ 1, 4 ]) ], [ 4, NF(5,[ 1, 4 ]) ],
> [ 1, NF(24,[ 1, 5, 7, 11 ]), 24, [ [ 2, 5, 18 ], [ 2, 7, 12 ] ], [ [ 6 ] ] ], [ 4, NF(5,[ 1, 4 ]) ],
> [ 1, NF(40,[ 1, 11, 29, 39 ]), 40, [ [ 2, 11, 20 ], [ 2, 29, 30 ] ], [ [ 10 ] ] ],
> [ 4, NF(15,[ 1, 14 ]) ], [ 1, NF(5,[ 1, 4 ]), 60, [ [ 2, 11, 0 ], [ 2, 19, 30 ], [ 2, 31, 30 ] ],
> [ [ 0, 45 ], [ 15 ] ] ] ]
> gap> A:=wd[27];
> [ 1, NF(5,[ 1, 4 ]), 60, [ [ 2, 11, 0 ], [ 2, 19, 30 ], [ 2, 31, 30 ] ], [ [ 0, 45 ], [ 15 ] ] ]
> gap> LocalIndicesOfCyclotomicAlgebra(A);
> [ [ infinity, 2 ] ]

I don't either on the current divalg branch.  It's on pr/72 where DefiningGroup functions use PermGroup in place of SpecialPcGroup the problem occurs for me.

**Pull request closed without merging**